src/engine/sparqlExpressions/ExistsExpression.h

```diff
@@ -64,7 +64,25 @@ class ExistsExpression : public SparqlExpression {
   // Used to extract `EXISTS` expressions from a general expression tree.
   bool isExistsExpression() const override { return true; }
 
+  // Return all the variables that are used in this expression.
+  std::span<const Variable> getContainedVariablesNonRecursive() const override {
+    return argument_.selectClause().getSelectedVariables();
+  }
+
+  // Select all visible variables of the `EXISTS` expression that are present in
+  // `variables`.
```

Comment:
Suggestion for the comment:
Set the `SELECT` of the argument of this exists expression to all the variables that are visible in the argument AND contained in `variables`.